core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -65,17 +67,54 @@
 import org.infinispan.interceptors.locking.ClusteringDependentLogic;
 import org.infinispan.metadata.Metadata;
 import org.infinispan.notifications.cachelistener.CacheNotifier;
+import org.infinispan.remoting.responses.UnsuccessfulResponse;
 import org.infinispan.remoting.transport.Address;
 import org.infinispan.statetransfer.OutdatedTopologyException;
 import org.infinispan.statetransfer.StateConsumer;
 import org.infinispan.statetransfer.StateTransferLock;
 import org.infinispan.statetransfer.StateTransferManager;
+import org.infinispan.topology.CacheTopology;
+import org.infinispan.util.concurrent.IsolationLevel;
 import org.infinispan.util.logging.Log;
 import org.infinispan.util.logging.LogFactory;
 import org.infinispan.xsite.statetransfer.XSiteStateConsumer;
 
 /**
  * Interceptor in charge with wrapping entries and add them in caller's context.
+ * <p>
+ * This interceptor loads the entry from {@link DataContainer} if this node is an owner of the key according to
+ * {@link CacheTopology#getReadConsistentHash()}. The entry is committed then through {@link ClusteringDependentLogic}
+ * and that logic checks {@link CacheTopology#getWriteConsistentHash()} if the entry can be committed. Entry being
+ * wrapped is a prerequisite for the command to run and therefore commit the entry, but it's not up to this interceptor
+ * to make sure the entry is always wrapped.
+ * <p>
+ * If the entry should be read locally but it's not found in DC, the entry will be wrapped (either as NullCacheEntry
+ * for reads or other appropriate type for writes). Such entry must then return <code>false</code> on
+ * {@link CacheEntry#skipLookup()} as it's value is unsure (subsequent interceptors can retrieve the new value from the
+ * cache store or remote node and call {@link EntryFactory#wrapExternalEntry} to update the context and make the entry valid).
+ * <p>
+ * With repeatable reads, the value that is context must not be overwritten by value out of the transaction (only commands
+ * in this transaction can change the context entry. That's why we need to call {@link CacheEntry#setSkipLookup} after
```

Comment:
Ok, I just took the comment to mean that with read-committed it's ok to re-fetch the value from a remote node.